src/SIL.LCModel.Core/SpellChecking/SpellEngine.cs

```diff
@@ -50,21 +63,30 @@ internal SpellEngine(string affixPath, string dictPath, string exceptionPath)
 			catch (Exception e)
 			{
 				Debug.WriteLine("Initializing Hunspell: {0} exception: {1} ", e.GetType(), e.Message);
+				#if __MonoCS__
 				if (m_hunspellHandle != IntPtr.Zero)
 				{
 					Hunspell_uninitialize(m_hunspellHandle);
 					m_hunspellHandle = IntPtr.Zero;
 					throw;
 				}
+				#else
+				if (m_hunspellHandle != null)
+				{
+					m_hunspellHandle.Dispose();
+					throw;
```

Comment:
Since we're working in the area, it would be nice to change these to `throw new Exception(e);` to preserve the original stacktrace